server/controllers/LikeController.js

```diff
@@ -0,0 +1,164 @@
+
+const { Article, ArticleLike, User } = db;
+
+/**
+ * This class contains static methods for liking and unliking
+ * articles in the applications. It also contains methods
+ * for retrieving like informations for a particular article
+ * @class LikeController
+ * @returns {object} when called with the  new it returns an
+ * instance of LikeController but this is never used
+ */
+class LikeController {
+  /**
+   * This function is a middleware and is called when the user wants
+   * to like a particular article.
+   * @param {*} req the request object containing the user's request
+   * @param {*} res the response object used to send a respons to the user
+   * @param {*} next called when an error occurs
+   * @returns {void} does not return a value
+   * @author Chidiebere Oguejiofor
+   */
+  static likeArticle(req, res, next) {
+    const { slug } = req.params;
+    const { decoded: user } = req.body;
+
+    Article.findOne({
+      where: { slug },
+      attributes: ['id', 'title', 'slug'],
+      include: {
+        model: User,
+        as: 'author',
+        attributes: ['id', 'firstName', 'lastName', 'image'],
+      },
+    }).then((article) => {
+      if (!article) {
+        return res.status(404).json({
+          status: 'error',
+          error: {
+            message: 'Article with that slug was not found',
+          },
+        });
+      }
+      const likeObj = {
+        userId: user.id,
+        articleId: article.id,
+      };
+      return ArticleLike.findOrCreate({
+        where: likeObj,
+        defaults: likeObj,
+        attributes: [['updatedAt', 'timeLiked']],
+      }).then(([, created]) => {
+        if (created) {
+          return res.status(201).json({
+            status: 'success',
+            data: article.dataValues,
+            message: 'Successfully liked the article',
+          });
+        }
+        return res.status(200).json({
+          status: 'success',
+          data: article.dataValues,
+          message: 'You had already liked this article',
+        });
+      });
+    }).catch(next);
+  }
+
+  /**
+   * This unlikes an article specified in the request.
+   * @param {object} req an object containing the request made to the server
+   * @param {object} res this object is used to send a response to the user
+   * @param {Function} next  called when an error occurs
+   * @author Chidiebere Oguejiofor
+   * @returns {void} does not return a value
+   */
+  static unlikeArticle(req, res, next) {
+    const { slug } = req.params;
+    const { decoded: user } = req.body;
+
+
+    Article.findOne({ where: { slug } })
+      .then((article) => {
+        if (article) {
+          ArticleLike.destroy({
+            where: {
+              userId: user.id,
+              articleId: article.id,
+            },
+          }).then((deleted) => {
+            res.status(deleted ? 200 : 400).json({
+              status: deleted ? 'success' : 'fail',
```

Comment:
NO need to return this `status`, and also, it can be confused with HTTP status code